src/Miso/Fetch.hs

```diff
@@ -160,41 +187,94 @@ instance (Fetch api, KnownSymbol name) => Fetch (QueryFlag name :> api) where
+      (ms $ reflectMethod (Proxy @method))
+      (options ^. body)
+      (options ^. headers <> [(ms "Accept", ms $ renderHeader $ contentType ctProxy)])
+      error_
+      success_
     where
-      method = ms (reflectMethod (Proxy @method))
-      params = MS.concat
-        [ mconcat
-           [ ms "?"
-           , MS.intercalate (ms "&")
-             [ k <> ms "=" <> v
-             | (k,v) <- options ^. queryParams
-             ]
-           ]
-         | not $ null (options ^. queryParams)
-         ]
-      flags = MS.mconcat [ ms "?" <> k | k <- options ^. queryFlags ]
-      url = options ^. baseUrl <> options ^. currentPath <> params <> flags
+      ctProxy = Proxy @ct
+instance {-# OVERLAPPING #-} (ReflectMethod method) => Fetch (Verb method code cts NoContent) where
+  type ToFetch (Verb method code cts NoContent) = (MisoString -> JSM ()) -> (NoContent -> JSM ()) -> JSM ()
+  fetchWith Proxy = fetchNoContent $ Proxy @method
+instance (ReflectMethod method) => Fetch (NoContentVerb method) where
+  type ToFetch (NoContentVerb method) = (MisoString -> JSM ()) -> (NoContent -> JSM ()) -> JSM ()
+  fetchWith Proxy = fetchNoContent $ Proxy @method
+fetchNoContent :: ReflectMethod method => Proxy method -> FetchOptions -> (MisoString -> JSM ()) -> (NoContent -> JSM ()) -> JSM ()
+fetchNoContent proxy_ options error_ success_ =
+    fetchFFI
+      (const $ pure NoContent)
+      (optionsToUrl options)
+      (ms $ reflectMethod proxy_)
+      (options ^. body)
+      (options ^. headers)
+      error_
+      success_
+-----------------------------------------------------------------------------
+instance
+  ( contentTypes ~ (ct ': otherContentTypes),
+    as ~ (a ': as'),
+    AllMime contentTypes,
+    ReflectMethod method,
+    All (UnrenderResponse contentTypes) as
+  ) =>
+  Fetch (UVerb method contentTypes as)
+  where
+  type ToFetch (UVerb method contentTypes as) = (MisoString -> JSM ()) -> (Union as -> JSM ()) -> JSM ()
+  fetchWith Proxy options error_ success_ =
+    fetchFFI
+      (first unlines . tryParsers . mimeUnrenders . MS.fromMisoString)
+      (optionsToUrl options)
+      (ms $ reflectMethod (Proxy @method))
+      (options ^. body)
+      (options ^. headers <> map ((ms "Accept",) . ms . renderHeader) (allMime $ Proxy @contentTypes))
+      error_
+      success_
+    where
+      mimeUnrenders :: BSL.ByteString -> NP ([] :.: Either (MediaType, String)) as
+      mimeUnrenders body_ = cpure_NP (Proxy @(UnrenderResponse contentTypes))
+        $ Comp $ unrenderResponse (Proxy @contentTypes) body_
+      tryParsers :: NP ([] :.: Either (MediaType, String)) xs -> Either [String] (Union xs)
+      tryParsers = \case
+        Nil -> Left ["no parsers"]
+        Comp x :* xs -> case partitionEithers x of
+          (err', []) -> bimap (map (\(t, s) -> show t <> ": " <> s) err' <>) S $ tryParsers xs
+          (_, (res : _)) -> Right . inject . I $ res
+class UnrenderResponse (cts :: [Type]) (a :: Type) where
```

Comment:
Should we use `AllMimeUnrender` w/ `MimeUnrender` from 

https://hackage.haskell.org/package/servant-0.20.2/docs/Servant-API-ContentTypes.html#t:MimeUnrender

Reply:
Yes, good point, actually we can! Just pushed as https://github.com/dmjio/miso/pull/888/commits/59b85a1474e322b4209180d8cac434dbecf79ed9.

I was copying `servant-client` a bit too closely here. I guess they're interesting enough to need their own type class for this, but we're not.